src/ProjNavigator/sources_form.cpp

```diff
@@ -280,9 +281,15 @@ void SourcesForm::SlotRemoveFile() {
   QTreeWidgetItem *item = m_treeSrcHierachy->currentItem();
   if (item == nullptr) return;
 
+  auto strFileName = item->text(0);
+  if (strFileName.contains(SRC_TREE_FLG_TOP))
+    strFileName.remove(SRC_TREE_FLG_TOP);
   auto fileSet = item->data(0, SetFileDataRole);
   if (!fileSet.isNull()) {
-    QString strFileName = item->text(0);
+    auto questionStr = tr("Should %1 file be removed?").arg(strFileName);
```

Comment:
@ravic-rs please correct text here

Reply:
Nice catch Skyler. Actually a dangerous thing is happening in ProjectFileSet::deleteFile:
for (auto it = m_langMap.begin(); it != m_langMap.end(); ++it) {
...
m_langMap.erase(it);
...
}

as that file is the last in the collection, it get invalidated and application crashes. I'll think on another way erasing it.

Reply:
I think the standard confirmation is "Are you sure" so maybe "Are you sure you want to remove %1 from the project?", I'm open to some other wording for "remove %1 from the project?" if you think there is something better.

Reply:
I'm ok with having it as your proposed.
"Are you sure you want to remote %1 from the project?"
A text below that informs user that we are not deleting the file from disk.
If the user can add this file back using a mechanism then we should provide how to do that as well.
On second thought, I am wondering if we should change the "Remove" to "Delete" instead. Since the above comments with help them understand the difference. The confusion will arise when the shortcut in menu is shown up as "Del"